app/components/machine/driver-aliyunecs/component.js

```diff
@@ -10,23 +10,585 @@ let ioOptimized=[
+    description: "ecs.i1-c10d1.8xlarge - 32 CPU,128 GB memory,6 Gbps network bandwidth,800k PPS",
+  },
+  {
+    value: "ecs.i1.14xlarge",
+    description: "ecs.i1.14xlarge - 56 CPU,224 GB memory,10 Gbps network bandwidth,1200k PPS",
+  },
+  {
+    value: "ecs.cm4.xlarge",
+    description: "ecs.cm4.xlarge - 4 CPU,16 GB memory,1.5 Gbps network bandwidth,200k PPS",
+  },
+  {
+    value: "ecs.cm4.2xlarge",
+    description: "ecs.cm4.2xlarge - 8 CPU,32 GB memory,3 Gbps network bandwidth,400k PPS",
+  },
+  {
+    value: "ecs.cm4.4xlarge",
+    description: "ecs.cm4.4xlarge - 16 CPU,64 GB memory,6 Gbps network bandwidth,800k PPS",
+  },
+  {
+    value: "ecs.cm4.6xlarge",
+    description: "ecs.cm4.6xlarge - 24 CPU,96 GB memory,10 Gbps network bandwidth,1200k PPS",
+  },
+  {
+    value: "ecs.ce4.xlarge",
+    description: "ecs.ce4.xlarge - 4 CPU,32 GB memory,1.5 Gbps network bandwidth,200k PPS",
+  },
+  {
+    value: "ecs.c4.xlarge",
+    description: "ecs.c4.xlarge - 4 CPU,8 GB memory,1.5 Gbps network bandwidth,200k PPS",
+  },
+  {
+    value: "ecs.c4.2xlarge",
+    description: "ecs.c4.2xlarge - 8 CPU,16 GB memory,3 Gbps network bandwidth,400k PPS",
+  },
+  {
+    value: "ecs.c4.4xlarge",
+    description: "ecs.c4.4xlarge - 16 CPU,32 GB memory,6 Gbps network bandwidth,800k PPS",
+  },
+];
+
 export default Ember.Component.extend(Driver, {
   driverName       : 'aliyunecs',
   aliyunecsConfig      : Ember.computed.alias('model.aliyunecsConfig'),
   ioOptimized : ioOptimized,
+  instanceType: instanceType,
+  dataDiskCategory: dataDiskCategory,
+  systemDiskCategory: systemDiskCategory,
+  regions: regions,
 
-
+  ioOptimizedObserves: function(){
+    if (this.get('aliyunecsConfig.ioOptimized')==='none'){
+      this.set('instanceType',notOptimizedinstanceType);
+      this.set('dataDiskCategory',[{value:"cloud"}]);
+      this.set('systemDiskCategory',[{value:"cloud"}]);
+    }
+    else {
+      this.set('instanceType',isOptimizedinstanceType);
+      this.set('dataDiskCategory',optimizedDiskCategory);
+      this.set('systemDiskCategory',optimizedDiskCategory);
+    }
+  }.observes('aliyunecsConfig.ioOptimized'),
   bootstrap: function() {
     let config = this.get('store').createRecord({
       type                  : 'aliyunecsConfig',
     });
-
+    this.set('instanceType',notOptimizedinstanceType);
     this.set('model', this.get('store').createRecord({
       type: 'host',
       aliyunecsConfig: config,
     }));
-
+    this.set('model.engineRegistryMirror',['https://s06nkgus.mirror.aliyuncs.com']);
+    this.set('model.engineInstallUrl','http://dev-tool.oss-cn-shenzhen.aliyuncs.com/docker-install/1.12.6.sh');
     //this.set('editing', false);
   },
+  validate: function(){
+    this._super();
+    var errors = this.get('errors') || [];
+    var name = this.get('model.hostname');
+    if (name) {
+      //name length rules
+      if (name.length<2){
+        errors.push("Name should have more then 2 letters.");
```

Comment:
These should all go into the translation files, particularly since you probably want to provide them in Chinese also...

`intl: Ember.inject.service(),` at the top, then

`errors.push(this.get('intl').t('driverAliyunecs.error.nameTooShort'));`